### PR TITLE
Add opt gene tutorial

### PR DESCRIPTION
**Pull request by snmendoz**
[*I added a tutorial for optGene and improved optgene related functions*]

**I hereby confirm that I have:**

- [X ] Tested my code on my own machine
- [ X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [ X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [ ] with [X] to check the box)*

### CODE REVIEW

**Comment by syarra** [comment]
@snmendoz I ran your tutorial and I get the following errors:
optgene
Could you, please, have a look at them and fix them? If you need any help, please ping me.

**Comment by snmendoz** [comment]
Hi @syarra. I got the same error when I was using glpk but I changed to gurobi and this issue was solved. What solver are you using? I think I'm going to use a simpler function to avoid this kind of problems.

**Comment by snmendoz** [comment]
@syarra, please try again. I fixed the function analyzeOptKnock.

regards,

File: src/design/GetOptGeneSol.m [hunk 1]
```diff
@@ -14,10 +14,14 @@
...
 %    isGeneList:      boolean
+%    saveFile:        boolean. Default = 0;
 %
 % OUTPUT:
 %    optGeneSol:      Solution in the desired format
 
+if nargin < 9 || isempty(saveFile)
+    saveFile = 0;s
+end
 writeDirect = 'C:\';
```
Comment (syarra) [inline src/design/GetOptGeneSol.m]: oh, this is dangerous ... Only compatible with Windows ....

Comment (snmendoz) [inline src/design/GetOptGeneSol.m]: I agree! I will change it

Comment (syarra) [inline src/design/GetOptGeneSol.m]: Thanks @snmendoz!

**Comment by syarra** [comment]
I tried again your tutorial with your fixes  and 🎉 it is working now (with Gurobi)!
I still have no luck with GLPK, though it runs without error message:
screen shot 2017-06-21 at 17 44 25


I guess you should enforce people to use Gurobi at the beginning of the tutorial. What do you think?

Could you explain me the meaning of the timings reported in the `TIMING` section?
The runs are about 6 minutes... What about the `(2-3 minutes for run)` statement?

Thanks @snmendoz for your good work!

**Comment by snmendoz** [comment]
Good idea. I will put a statement encouraging people to use gurobi. 

optGene takes ~2 minutes each time it is executed (2 minutes per run), so since optGene is executed three times, the total time is 6 minutes. I will change it to 6 minutes, so peple will know the total time.

**Comment by snmendoz** [comment]
@syarra all the pending tasks were performed. 

Regards,

**Comment by syarra** [comment]
Thanks a lot @snmendoz for your contributions! Good job!

**Pull request merged**